include/walnuts/util.hpp

```diff
@@ -273,7 +273,17 @@ class NoExceptLogpGrad {
     }
   }    
 
-  F& logp_grad_;
+  /**
+   * @brief Return the number of log density/gradient calls.
+   *
+   * @return The number of calls.
+   */
+  Integer logp_grad_calls() const noexcept {
+    return calls_;
+  }
+  
+  const F& logp_grad_;
+  Integer calls_;
```

Comment:
I don’t think we necessarily should track this in this object. It seems simple enough for users who want it to do themselves, and it’s slightly annoying to have to remove all those const qualifiers as a result